javascript/javascript3/week3/TASK3-Shopping cart/index.js

```diff
@@ -1,149 +1,142 @@
-        const tempProducts = this.products.map((product) => product);
-        // remove item from temporary array
-        tempProducts.splice(indexOfProduct, 1);
-        // replace products with the temporary array
-        this.products = tempProducts;
-      } else {
-        console.log("Can't find item.");
-      }
-    }
+  addProduct(product) {
+    this.products.push(product);
+  }
 
-    searchProduct(productName) {
-      return this.products.filter((product) =>
-        product.name.toLowerCase().includes(productName.toLowerCase()),
-      );
+  removeProduct(product) {
+    const indexOfProduct = this.products.indexOf(product);
+    if (indexOfProduct > -1) {
+      this.products.splice(indexOfProduct, 1);
+    } else {
+      console.log("Can't find item.");
     }
+  }
 
-    getTotal() {
-      return this.products.reduce((result, currentProduct) => result + currentProduct.price, 0);
-    }
+  searchProduct(productName) {
+    return this.products.filter((product) =>
+      product.name.toLowerCase().includes(productName.toLowerCase()),
+    );
+  }
 
-    renderProducts() {
-      const fragment = new DocumentFragment();
-      const ul = document.createElement("UL");
-      fragment.appendChild(ul);
-      this.products.forEach((product) => {
-        createListElement(product);
-      });
-
-      function createListElement(productToList) {
-        let mainLI = document.createElement("LI");
-        ul.appendChild(mainLI);
-        let productUL = document.createElement("UL");
-        mainLI.appendChild(productUL);
-        let nameLI = document.createElement("LI");
-        nameLI.textContent = `Procuct: ${productToList.name}`;
-        productUL.append(nameLI);
-        let priceLI = document.createElement("LI");
-        priceLI.textContent = `Price: ${productToList.price}`;
-        productUL.append(priceLI);
-        document.body.appendChild(fragment);
-      }
-    }
+  getTotal() {
+    return this.products.reduce((result, currentProduct) => result + currentProduct.price, 0);
+  }
 
-    getUser = async (userID) => {
-      try {
-        const baseUrl = "https://jsonplaceholder.typicode.com/users/";
-        const query = userID;
-        return await getData(baseUrl, query);
-      } catch (error) {
-        console.log(error);
-      }
-    };
+  renderProducts() {
+    const fragment = new DocumentFragment();
+    const ul = document.createElement("UL");
+    fragment.appendChild(ul);
+    this.products.forEach((product) => {
+      createListElement(product);
+    });
   }
 
-  getData = async (baseUrl, query) => {
+  async getUser(userID) {
     try {
-      const response = await fetch(`${baseUrl}${query}`);
-      if (response.status >= 200 && response.status < 300) {
-        const data = response.json();
-        return data;
-      } else {
-        throw new Error(`HTTP error! status: ${response.status}`);
-      }
+      const BASE_URL = "https://jsonplaceholder.typicode.com/users/";
```

Comment:
even better, take this constant OUT of the function. Then you don't have to pass it around.